### PR TITLE
Refactored

### PR DESCRIPTION
**Pull request by TongilKim**
Refactored the class component to function component

### CODE REVIEW

**Comment by shawnohn** [comment]
Everything looks find except App is still class component.

**Comment by TongilKim** [comment]
I refixed the project because of the login status issue.

**Comment by TongilKim** [comment]
> Everything looks find except App is still class component.

- I'm sorry I just checked this reply, so I tried to switch the class to function component but if I changed it, the loggedIn state is not working properly. I will figure it out ASAP :)

File: client/src/components/Login.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+            return;
+        }
+        setLoginErr(false);
+    };
+
+    const validate = () => {
+        let emailErr = "";
+        let passwordErr = "";
+
+        if (!email.includes("@")) {
```
Comment (sina-jamshidi) [inline client/src/components/Login.js:77]: Not a big deal but maybe using something like the [answer here](https://stackoverflow.com/questions/46155/how-to-validate-an-email-address-in-javascript) for email validation?

File: client/src/components/SignUp.js [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+import MuiAlert from '@material-ui/lab/Alert';
+import axios from 'axios';
+import { Redirect } from 'react-router-dom';
+
+const useStyles = theme => ({
+    form: {
+        display: 'flex',
+        flexDirection: 'column',
+        margin: 'auto',
+        // width: 'fit-content',
```
Comment (sina-jamshidi) [inline client/src/components/SignUp.js:15]: commented out code

File: client/src/components/SignUp.js [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+            /////////// START here to store the data in the back//////////
+            const user = {
+                username: name,
+                userEmail: email,
+                password: password
+            }
+            //console.log(user);
+            axios.post('/users/add', user)
+                .then(response => {
+                    // console.log(response)
```
Comment (sina-jamshidi) [inline client/src/components/SignUp.js:62]: commented out code

File: server/.env [hunk 4]
```diff
@@ -1,2 +1,4 @@
 TEAM_MEMBERS="Aecio, Bonnie, Shums, Pray, Shawn, Tongil"
 DATABASE_CONNECT="mongodb+srv://team-kiwi-user:hatchways-team-kiwi@team-kiwi-cluster-yv3im.mongodb.net/hatchways?retryWrites=true&w=majority"
+ACCESS_TOKEN_LOGIN= "7186dd676fb617a4500de62d84a71e7976a47e5f0953b2bc6c1fc65544bde024e1d782e5e73a877cc820fd6e71386a7effe255f70613e5658c54496a538dd78b"
```
Comment (sina-jamshidi) [inline server/.env:3]: this file shouldn't be getting committed

File: server/passport/localStrategy.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+const User = require('../models/userModel')
+const LocalStrategy = require('passport-local').Strategy
+
+const strategy = new LocalStrategy(
+	{
+		usernameField: 'userEmail' // not necessary, DEFAULT
+	},
+	function (userEmail, password, done) {
+		//console.log("In localStrategy", userEmail, password)
```
Comment (sina-jamshidi) [inline server/passport/localStrategy.js:9]: commented out code

**Review by sina-jamshidi — APPROVED** [review]
Looks good to me. Just a few minor changes.

**Comment by sina-jamshidi** [comment]
> > Everything looks find except App is still class component.
> 
> * I'm sorry I just checked this reply, so I tried to switch the class to function component but if I changed it, the loggedIn state is not working properly. I will figure it out ASAP :)

If it turns out too be too much trouble don't worry about it for now. We can revisit it if we have time later in the project

File: server/.env (hunk 4 above)
Comment (shawnohn) [reply server/.env:3]: Please refer this.
https://www.freecodecamp.org/news/gitignore-what-is-it-and-how-to-add-to-repo/

**Pull request merged**